src/main/python/org/broadinstitute/hellbender/gcnvkernel/postprocess/viterbi_segmentation.py

```diff
@@ -0,0 +1,371 @@
+            io_commons.assert_files_are_identical(model_calling_config_file, calls_calling_config_file)
+
+            # extract and store sample names for the current shard
+            scattered_sample_names.append(ViterbiSegmentationEngine._get_sample_names_from_calls_shard(calls_path))
+
+        # all scattered calls have the same set of samples and in the same order
+        assert len(set(scattered_sample_names)) == 1,\
+            "The calls shards contain different sample names and/or different number of samples."
+
+        # all samples have ploidy calls in the metadata collection
+        sample_names = list(scattered_sample_names[0])
+        sample_metadata_collection.all_samples_have_ploidy_metadata(sample_names)
+
+    @staticmethod
+    def _get_sample_names_from_calls_shard(calls_path: str) -> Tuple[str]:
+        sample_names: Tuple[str] = ()
+        sample_index = 0
+        while True:
+            sample_posteriors_path = io_denoising_calling.get_sample_posterior_path(calls_path, sample_index)
+            if not os.path.isdir(sample_posteriors_path):
+                break
+            sample_names += (io_commons.get_sample_name_from_txt_file(sample_posteriors_path),)
+            sample_index += 1
+        if len(sample_names) == 0:
+            raise Exception("Could not file any sample posterior calls in {0}.".format(calls_path))
+        else:
+            return tuple(sample_names)
+
+    @staticmethod
+    def _get_denoising_config(input_path: str) -> DenoisingModelConfig:
+        return DenoisingModelConfig.from_json_file(os.path.join(
+            input_path, io_consts.default_denoising_config_json_filename))
+
+    @staticmethod
+    def _get_calling_config(input_path: str) -> CopyNumberCallingConfig:
+        return CopyNumberCallingConfig.from_json_file(os.path.join(
+            input_path, io_consts.default_calling_config_json_filename))
+
+    @staticmethod
+    def _get_interval_list_from_model_shard(model_path: str) -> List[Interval]:
+        interval_list_file = os.path.join(model_path, io_consts.default_interval_list_filename)
+        return io_intervals_and_counts.load_interval_list_tsv_file(interval_list_file)
+
+    @staticmethod
+    def _get_log_q_tau_tk_from_model_shard(model_path: str) -> np.ndarray:
+        return io_commons.read_ndarray_from_tsv(os.path.join(
+            model_path, io_consts.default_class_log_posterior_tsv_filename))
+
+    @staticmethod
+    def _get_log_copy_number_emission_tc_from_calls_shard(calls_path: str, sample_index: int):
+        return io_denoising_calling.SampleDenoisingAndCallingPosteriorsImporter.\
+            import_ndarray_tc_with_copy_number_header(
+                io_denoising_calling.get_sample_posterior_path(calls_path, sample_index),
+                io_consts.default_copy_number_log_emission_tsv_filename)
+
+    @staticmethod
+    def coalesce_seq_into_segments(seq: List[TypeVar('_T')]) -> List[Tuple[TypeVar('_T'), int, int]]:
```

Comment:
I also don't think this method needs to be exposed.  Does it work OK for lists of length 1 (I didn't check)?

Reply:
I hid it. Also, it does work for length 1.